### ISSUE TITLE
Update CDDL generation script to also read external definitions

### ISSUE THREAD

**Issue by whimboo** [w3c/webdriver-bidi#791]
We have external definitions of WebDriver BiDi command and events like this on-exhaustive list shows:

https://w3c.github.io/webdriver-bidi/#external-specifications

Right now we do not yet read their CDDL defintions which means even if implemented in browsers clients will not know about in case they rely on auto-generated code.

We should probably update our cddl generation script to check external documents as well.

CC @jgraham @gsnedders

**Comment by gsnedders** [comment]
We should be careful to not accidentally change the definition of the [Remote End Definition](https://w3c.github.io/webdriver-bidi/#remote-end-definition) (or Local End Definition) if/when we do this; various things are matched against it, and if we explicitly include all external specifications in the definition, does that not oblige them to be implemented?

**Comment by whimboo** [comment]
The inclusion of any external spec would require an agreement by the Browser Testing and Tools group for sure. Further it would apply to any other CDDL changes afterward. 

If we don't agree on that what else could we do to provide a CDDL export that also includes all the extension commands and events?

**Comment by tidoust** [comment]
I haven't had time to dig into it yet but I plan to work on adding CDDL extracts to Webref data in a not so distant future, see https://github.com/w3c/webref/issues/1353.

There are also interesting considerations on the notion of CDDL modules in a [recent IETF thread](https://mailarchive.ietf.org/arch/msg/tools-discuss/-oa5-sKE6OdRPx63f9KaQJHmVlE/) (initially focused on ABNF extracts), which could perhaps be used to create a consolidated view.

**Comment by tidoust** [comment]
Webref now has [CDDL extracts](https://github.com/w3c/webref/tree/curated/ed/cddl) for individual specifications (for WebDriver BiDi related ones, one extract for the remote end definition, one for the local end definition, and one that contains all definitions). Extracts in the `curated` branch are guaranteed to be valid CDDL.

What's missing is creating a merged view of WebDriver BiDi related CDDL extracts from the list in [External specifications](https://w3c.github.io/webdriver-bidi/#external-specifications).

It would be easy to merge relevant CDDL extracts from Webref automatically, starting from the WebDriver BiDi one, but something seems missing: the extension modules currently do not link back to the core CDDL constructs defined in WebDriver BiDi. For example, taking the example of [Web Bluetooth](https://webbluetoothcg.github.io/web-bluetooth/#automated-testing), I would expect that spec to define something like:

```cddl
CommandData //= BluetoothCommand
EventData //= BluetoothEvent
ErrorCode /= "no such device" / "no such prompt"
```

Without these definitions, nothing relates the CDDL defined in Web Bluetooth with the one in WebDriver BiDi. Or do I miss something? Also, are these the right extension points?

In other words, in parallel to merging the CDDL extracts, CDDL in the extension specs would also benefit from being updated to make the result useful.

(Note: `CommandData`, `EventData` and `ErrorCode` definitions would better be exported so that other specs can reference them).
